cli/src/cmd/forge/cache.rs

```diff
@@ -0,0 +1,106 @@
+//! cache command
+
+use clap::{Parser, Subcommand};
+use std::str::FromStr;
+
+use crate::cmd::Cmd;
+use ethers::prelude::Chain;
+use eyre::Result;
+use foundry_config::Chain as FoundryConfigChain;
+use foundry_config::Config;
+
+#[derive(Debug, Parser)]
+pub struct CacheArgs {
+    #[clap(subcommand)]
+    pub sub: CacheSubcommands,
+}
+
+#[derive(Debug)]
+pub enum ChainOrAll {
+    Chain(Chain),
+    All,
+}
+
+impl FromStr for ChainOrAll {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Ok(chain) = ethers::prelude::Chain::from_str(s) {
+            Ok(ChainOrAll::Chain(chain))
+        } else if s == "all" {
+            Ok(ChainOrAll::All)
+        } else {
+            Err(format!("Expected known chain or all, found: {s}"))
+        }
+    }
+}
+
+#[derive(Debug, Parser)]
+pub struct CleanArgs {
+    // TODO refactor to dedup shared logic with ClapChain in opts/mod
+    #[clap(
+        env = "CHAIN",
+        default_value = "all",
+        possible_values = [
+            "all",
+            "mainnet",
+            "ropsten",
+            "rinkeby",
+            "goerli",
+            "kovan",
+            "xdai",
+            "polygon",
+            "polygon_mumbai",
+            "avalanche",
+            "avalanche_fuji",
+            "sepolia",
+            "moonbeam",
+            "moonbeam_dev",
+            "moonriver",
+            "optimism",
+            "optimism-kovan"
```

Comment:
I think this is fine for now, wonder if clap allows this to be a function instead